platform/src/main/java/org/hillview/main/BatchLogAnalysis.java

```diff
@@ -0,0 +1,188 @@
+package org.hillview.main;
+
+import org.hillview.dataset.LocalDataSet;
+import org.hillview.dataset.api.Empty;
+import org.hillview.dataset.api.IDataSet;
+import org.hillview.dataset.api.IMap;
+import org.hillview.maps.ExtractValueFromKeyMap;
+import org.hillview.maps.FindFilesMapper;
+import org.hillview.maps.LoadFilesMapper;
+import org.hillview.sketches.*;
+import org.hillview.storage.FileSetDescription;
+import org.hillview.storage.IFileReference;
+import org.hillview.table.api.ITable;
+import org.hillview.utils.Converters;
+import org.hillview.utils.JsonList;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.cli.*;
+
+
+public class BatchLogAnalysis {
+
+    private static class HeatmapData {
+        long[][] matrix;
+        JsonList<String> errorCodeLabels = new JsonList<>();
+        ArrayList<String> timeLabels = new ArrayList<>();
+    }
+
+    private static HeatmapData heatmapErrTime(FileSetDescription desc, LocalDataSet<Empty> local, int numOfTimestampBuckets) {
```

Comment:
You should also document this function, even if it is private.
You should also add comments before each group of statements describing what they are doing. This will make reading the code easier.